### ISSUE TITLE
Incorrect nimbus value results in sparse submit exit without error message.

### ISSUE THREAD

**Issue by rduplain** [Parsely/streamparse#158]
`sparse submit` exits without error message, only usage, if nimbus is set incorrectly to a non-empty value in config.json.

If I run the wordcount quickstart:

```
sparse quickstart wordcount
cd wordcount
sparse submit
```

I get a meaningful error: `error: No Nimbus server configured in config.json.`

When I set `"nimbus": "example"` in config.json, `sparse submit` shows the usage message without any error message:

```
usage: sparse [-h]
              {jar,kill,list,quickstart,remove_logs,run,stats,submit,tail,update_virtualenv,visualize,worker_uptime}
              ...

Utilities for managing Storm/streamparse topologies.
...
```

This raises questions about streamparse's config.json loading. It's a lazily loaded config, where values are pulled as needed via `get_env_config` in the source. We could consider a configuration object which loads all configuration early and validates it, also pointing out unknown keywords which were provided. This approach would give clear feedback to the user as to whether config.json is valid.

I am using sparse v2.0.2 on Python 2.7.10, source-compiled on Ubuntu 14.04 64-bit.

**Comment by rduplain** [comment]
More specifically, this happens when the workers list is unset. I can get `sparse submit` to do work with these changes to the quickstart config.json:

```diff
7,9c7,9
<             "user": "",
<             "nimbus": "",
<             "workers": [],
---
>             "user": "rduplain",
>             "nimbus": "localhost",
>             "workers": ["localhost"],
16c16
<             "virtualenv_root": ""
---
>             "virtualenv_root": "./"
```

**Comment by dan-blanchard** [comment]
> We could consider a configuration object which loads all configuration early and validates it, also pointing out unknown keywords which were provided.

That sounds like a reasonable idea.

**Comment by dan-blanchard** [comment]
We were inadvertently catching all AttributeErrors for a little while. I fixed that in 16149563683e2ef6f76a2366460076ea9e55b9d1
